### PR TITLE
Do not change modality when blurring browser window in useFocusVisible

### PR DESCRIPTION
**Pull request by BartoszGamza**
Closes  #1912 

Added a watcher for window blur preventing modality change after returning to browser window. Similarly as I did in #1845. This should keep all the functionalities in place.

Had to tweak tests a bit since `render()` was not clearing the state of hook properly.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Easy to reproduce on docs of useFocusVisible:
- click anywhere to set pointer modality
- click outside browser window
- click on page again

## 🧢 Your Project:

[LiveChat](https://www.livechat.com/) Chat Widget

### CODE REVIEW

**Review by snowystinger — COMMENTED** [review]
Thank you for looking into this!!
Tested in Chrome, it appears to be working there, the only oddity I've found is:
Click a button in storybook actiongroup story
Ctrl+Tab to change to a new tab
Ctrl+Shift+Tab to change back
It gets keyboard focus on release of the Ctrl key

Tested in Safari, same thing as Chrome

Tested in Firefox, no issues

We do have some logic for ignoring modifier keys https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/interactions/src/useFocusVisible.ts#L62

Maybe we should add to the check for something like `!(e.keyUp === true && e.key === 'Control')`

**Comment by BartoszGamza** [comment]
Hi again! Thanks for such detailed check. I reproduced this behaviour and managed to fix it with also checking for shift key. WDYT? 
Let me know if it works for you 🚀

**Review by snowystinger — DISMISSED** [review]
Haha, thanks for working with my psuedocode. I've retested and it looks to be working.
Thanks for coming in with tests as well, much appreciated!
Approving, will get another team member to look at it as well.

**Comment by BartoszGamza** [comment]
@snowystinger I see that some tests failed after merging main branch, should we worry?

**Comment by snowystinger** [comment]
@BartoszGamza yep, we should worry. Both test suites for React 16 and 17 failed in the same way, so it's probably not a random failure/flakey test.
If you have time to look at the test, that's great, otherwise I'll try to look at it in the near future when I have some time.

File: packages/@react-aria/interactions/src/useFocusVisible.ts [hunk 1]
```diff
@@ -106,7 +108,7 @@ function handleFocusEvent(e: FocusEvent) {
 
   // If a focus event occurs without a preceding keyboard or pointer event, switch to virtual modality.
   // This occurs, for example, when navigating a form with the next/previous buttons on iOS.
-  if (!hasEventBeforeFocus && !hasChangedTabRecently()) {
+  if (!hasEventBeforeFocus && !hasBlurredWindowRecently && !hasChangedTabRecently()) {
```
Comment (devongovett) [inline packages/@react-aria/interactions/src/useFocusVisible.ts:111]: Is `hasChangedTabRecently` still needed then? I assume the blur event is also fired when switching tabs?

Comment (BartoszGamza) [reply packages/@react-aria/interactions/src/useFocusVisible.ts:111]: Actually, yes! I have checked it for all browsers and it works fine, will adjust the solution 👍

**Comment by BartoszGamza** [comment]
@devongovett I have removed the handling of `hasChangedTabRecently` as you correctly noticed it is not needed.

@snowystinger I have investigated the issue with tests and it is indeed valid, related to drag&drop components. Thing is, the modality in tested case is treated as `keyboard` and not as `virtual`:

Screenshot 2021-05-20 at 12 30 57

Screenshot 2021-05-20 at 12 31 09

It all most probably narrows down to [useDragModality](https://github.com/adobe/react-spectrum/blob/366664b584c7af2a06343c2747907f7b7a148a34/packages/%40react-aria/dnd/src/utils.ts#L47) but Im not sure what can be done about this at this point. @devongovett , as you are the author, would you mind taking a look at this or maybe explaining a bit reasoning behind decisions made here (why we treat certain inputs as virtual)? 

I will come back to the problem tomorrow or at the beginning of next week. Thank you in advance 🙏

**Comment by devongovett** [comment]
@BartoszGamza the problem was that the tests for drag and drop were triggering a blur on the window as a way of reseting the focus visibility state. This used to work because then on the next focus it would be set to virtual. But now, on the next focus nothing will happen because a blur on the window occurred recently. I've worked around this by firing focus on `document.body` instead, which has the same effect.

**Review by devongovett — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**